src/main/java/com/fasterxml/jackson/databind/deser/std/StdDeserializer.java

```diff
@@ -1421,13 +1421,16 @@ protected final String _parseString(JsonParser p, DeserializationContext ctxt,
             return ob.toString();
         // 29-Jun-2020, tatu: New! "Scalar from Object" (mostly for XML)
         case JsonTokenId.ID_START_OBJECT:
+            act = _checkObjectToStringCoercion(p, ctxt, rawTargetType);
```

Comment:
Ideally should not be needed either, BUT may need to verify how "ctxt.extractScalarFromObject()" works, so...